### ISSUE TITLE
Attempt to generate some stylesheets from nvim treesitter files

### ISSUE THREAD

**Issue by foxfriends** [foxfriends/syncat#13]
> The problem with just adding a language to the language.toml is that there will still be no highlighting until someone makes a stylesheet

I see. Is it possible to generate sytlesheet from <https://github.com/nvim-treesitter/nvim-treesitter/tree/master/queries>?

_Originally posted by @Freed-Wu in https://github.com/foxfriends/syncat/issues/12#issuecomment-1783170520_

**Comment by foxfriends** [comment]
I have no plan to do this currently. You're welcome to try :)

**Comment by exoess** [comment]
hmmmm, well, i can start working on it, have you written like a full spec for the syncat style stuff, i'm working on translating the ecma/javascript query, but i'm stuck because some rules seem to be overriding others

**Comment by foxfriends** [comment]
https://github.com/foxfriends/syncat-themes?tab=readme-ov-file#creating-themes

I believe the most comprehensive documentation for syncat themes is in there^

**Comment by mister-choo** [comment]
Spend a couple hours on this. This is a hard problem. 
1. As mentioned before - nvim queries are laid out like this: less specific > more specific. 
Syncat styles work the opposite way. More specific highlights should go first
2. Secondly, parsing the queries is very non-trivial. My approaches with simple regexes didn't work. Looks like it needs a dedicated parser.

Was there a good reason to have a bespoke solution?

**Comment by foxfriends** [comment]
> Was there a good reason to have a bespoke solution?

not particularly, syncat was just something to build for fun. more fun to build something than to read docs on someone else's project

**Comment by mister-choo** [comment]
> > Was there a good reason to have a bespoke solution?
> 
> not particularly, syncat was just something to build for fun. more fun to build something than to read docs on someone else's project

Fair enough. A very good idea though. Shame it's not more actively developed

**Comment by Freed-Wu** [comment]
I write another [tool](https://texrocks.readthedocs.io/en/latest/topics/texcat.md.html) to syntax highlight. It uses neovim's tree-sitter's `queries/parser_name/highlights.scm` to capture scope name for nodes and map scope name to RGB color by VS Code color theme extension's theme XML/json files without recreate a new color theme.
